src/main/java/gregtech/api/recipes/ModHandler.java

```diff
@@ -626,7 +626,7 @@ public static boolean removeFurnaceSmelting(ItemStack input) {
 
         boolean wasRemoved = false;
         for (ItemStack stack : FurnaceRecipes.instance().getSmeltingList().keySet()) {
-            if (ItemStack.areItemStacksEqual(input, stack)) {
+            if ((stack.getItem() == input.getItem() && (stack.getMetadata() == GTValues.W || stack.getMetadata() == input.getMetadata()))) {
                 FurnaceRecipes.instance().getSmeltingList().remove(stack);
```

Comment:
Instead of simply calling `remove` after the correct check, use `FurnaceRecipes.instance().getSmeltingList().keySet().removeIf(stack -> (insert check here))`. Using map's default `removee` will just do the usual identity hash checkup which will fail.